src/test/java/net/snowflake/client/jdbc/ArrowBatchesIT.java

```diff
@@ -242,4 +257,272 @@ public void testDecimalBatch() throws Exception {
 
     assertEquals(3, totalRows);
   }
+
+  @Test
+  public void testVarCharBatch() throws Exception {
+    Statement statement = connection.createStatement();
+    ResultSet rs =
+        statement.executeQuery(
+            "select 'Gallia est ' union select 'omnis divisa ' union select 'in partes tres';");
+    ArrowBatches batches = rs.unwrap(SnowflakeResultSet.class).getArrowBatches();
+
+    int totalRows = 0;
+    List<Text> values = new ArrayList<>();
+
+    while (batches.hasNext()) {
+      ArrowBatch batch = batches.next();
+      List<VectorSchemaRoot> roots = batch.fetch();
+      for (VectorSchemaRoot root : roots) {
+        totalRows += root.getRowCount();
+        assertTrue(root.getVector(0) instanceof VarCharVector);
+        VarCharVector vector = (VarCharVector) root.getVector(0);
+        for (int i = 0; i < root.getRowCount(); i++) {
+          values.add(vector.getObject(i));
+        }
+        root.close();
+      }
+    }
+    assertNoMemoryLeaks(rs);
+    rs.close();
+
+    List<Text> expected =
+        new ArrayList<Text>() {
+          {
+            add(new Text("Gallia est "));
+            add(new Text("omnis divisa "));
+            add(new Text("in partes tres"));
+          }
+        };
+
+    assertTrue(values.containsAll(expected));
+
+    assertEquals(3, totalRows);
+  }
+
+  private class Pair<A, B> {
```

Comment:
Doesn't class SFPair work for you?

Reply:
Looking back on it, I decided agains it because I wasn't able to call a constructor, not realising there is a `public static of` method. My bad, will fix this